src/App.tsx

```diff
@@ -15,35 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+  default = '',
+}
+
+type SortObj = {
+  sortField: SortType;
+  isReversed?: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed = false }: SortObj,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.alphabetically:
+        return good1.localeCompare(good2);
+
+      case SortType.byLength:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
```

Comment:
The function name `getPreparedGoods` could be more descriptive if it started with a verb, reflecting what the function does. Consider renaming it to `prepareGoods` or `sortAndReverseGoods` to better align with the [CODE STYLE] guidelines.